digraph "unitGraph" {
    "r6 := @parameter0: java.lang.String[]"
    "r0 = newarray (int)[10]"
    "$r1 = new compiler.c1.Test8011706"
    "specialinvoke $r1.<compiler.c1.Test8011706: void <init>()>()"
    "r2 = $r1"
    "i0 = 0"
    "goto [?= (branch)]"
    "r2.<compiler.c1.Test8011706: int[] array> = null"
    "virtualinvoke r2.<compiler.c1.Test8011706: void m(boolean,int[],int[])>(1, r0, r0)"
    "i0 = i0 + 1"
    "if i0 < 10000 goto r2.<compiler.c1.Test8011706: int[] array> = null"
    "goto [?= $r3 = <java.lang.System: java.io.PrintStream out>]"
    "$r4 := @caughtexception"
    "r7 = $r4"
    "$r5 = new java.lang.RuntimeException"
    "specialinvoke $r5.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>(\"TEST FAILED\", r7)"
    "throw $r5"
    "$r3 = <java.lang.System: java.io.PrintStream out>"
    "virtualinvoke $r3.<java.io.PrintStream: void println(java.lang.String)>(\"TEST PASSED\")"
    "return"
    "r6 := @parameter0: java.lang.String[]"->"r0 = newarray (int)[10]";
    "r0 = newarray (int)[10]"->"$r1 = new compiler.c1.Test8011706";
    "$r1 = new compiler.c1.Test8011706"->"specialinvoke $r1.<compiler.c1.Test8011706: void <init>()>()";
    "specialinvoke $r1.<compiler.c1.Test8011706: void <init>()>()"->"r2 = $r1";
    "r2 = $r1"->"i0 = 0";
    "r2 = $r1"->"$r4 := @caughtexception";
    "i0 = 0"->"goto [?= (branch)]";
    "i0 = 0"->"$r4 := @caughtexception";
    "goto [?= (branch)]"->"if i0 < 10000 goto r2.<compiler.c1.Test8011706: int[] array> = null";
    "goto [?= (branch)]"->"$r4 := @caughtexception";
    "r2.<compiler.c1.Test8011706: int[] array> = null"->"virtualinvoke r2.<compiler.c1.Test8011706: void m(boolean,int[],int[])>(1, r0, r0)";
    "r2.<compiler.c1.Test8011706: int[] array> = null"->"$r4 := @caughtexception";
    "virtualinvoke r2.<compiler.c1.Test8011706: void m(boolean,int[],int[])>(1, r0, r0)"->"i0 = i0 + 1";
    "virtualinvoke r2.<compiler.c1.Test8011706: void m(boolean,int[],int[])>(1, r0, r0)"->"$r4 := @caughtexception";
    "i0 = i0 + 1"->"if i0 < 10000 goto r2.<compiler.c1.Test8011706: int[] array> = null";
    "i0 = i0 + 1"->"$r4 := @caughtexception";
    "if i0 < 10000 goto r2.<compiler.c1.Test8011706: int[] array> = null"->"goto [?= $r3 = <java.lang.System: java.io.PrintStream out>]";
    "if i0 < 10000 goto r2.<compiler.c1.Test8011706: int[] array> = null"->"r2.<compiler.c1.Test8011706: int[] array> = null";
    "if i0 < 10000 goto r2.<compiler.c1.Test8011706: int[] array> = null"->"$r4 := @caughtexception";
    "goto [?= $r3 = <java.lang.System: java.io.PrintStream out>]"->"$r3 = <java.lang.System: java.io.PrintStream out>";
    "$r4 := @caughtexception"->"r7 = $r4";
    "r7 = $r4"->"$r5 = new java.lang.RuntimeException";
    "$r5 = new java.lang.RuntimeException"->"specialinvoke $r5.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>(\"TEST FAILED\", r7)";
    "specialinvoke $r5.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>(\"TEST FAILED\", r7)"->"throw $r5";
    "$r3 = <java.lang.System: java.io.PrintStream out>"->"virtualinvoke $r3.<java.io.PrintStream: void println(java.lang.String)>(\"TEST PASSED\")";
    "virtualinvoke $r3.<java.io.PrintStream: void println(java.lang.String)>(\"TEST PASSED\")"->"return";
}
